mne/epochs.py

```diff
@@ -3543,31 +3544,34 @@ def equalize_epoch_counts(epochs_list, method="mintime"):
     --------
     >>> equalize_epoch_counts([epochs1, epochs2])  # doctest: +SKIP
     """
-    if not all(isinstance(e, BaseEpochs) for e in epochs_list):
+    if not all(isinstance(epoch, (BaseEpochs, EpochsTFR)) for epoch in epochs_list):
         raise ValueError("All inputs must be Epochs instances")
 
     # make sure bad epochs are dropped
-    for e in epochs_list:
-        if not e._bad_dropped:
-            e.drop_bad()
-    event_times = [e.events[:, 0] for e in epochs_list]
-    indices = _get_drop_indices(event_times, method)
-    for e, inds in zip(epochs_list, indices):
-        e.drop(inds, reason="EQUALIZED_COUNT")
-
-
-def _get_drop_indices(event_times, method):
+    for epoch in epochs_list:
+        # for EpochsTFR we can set _bad_dropped to True
+        if isinstance(epoch, EpochsTFR):
+            epoch._bad_dropped = True
+        if not epoch._bad_dropped:
+            epoch.drop_bad()
+    event_indices = [epoch.events[:, 0] for epoch in epochs_list]
+    indices = _get_drop_indices(event_indices, method)
+    for epoch, inds in zip(epochs_list, indices):
+        epoch.drop(inds, reason="EQUALIZED_COUNT")
+
+
+def _get_drop_indices(event_indices, method):
```

Comment:
ditto here; if you're going to change the parameter name let's change it to `sample_nums`.  But please also run `git grep _get_drop_indices` in your terminal to make sure that this file is the only place that the helper function is being called --- if not, we need to make sure that in the other places it's not being called with named parameters (if it is, then changing the parameter name here will cause those other usages to fail)